# pose config file for static poses (e.g. neutral)

# basic config
# sequence_velocity: 2.0
sequence_time: 0.8

# the neutral pose is a natural standing pose
# retracted pose must be preceded by the neutral pose
poses:
  - neutral
  - sit
  - retracted

# static sequences to be executed
sequences:
  - stand
  - retract
  - roll

# stand sequence should be used to stand up with the feet off the ground
stand:
  - sit
  - neutral

# retract sequence should be used to sit down and fold the limbs
retract:
  - sit
  - retracted

# roll sequence should be used to roll over
roll:
  - splayed
  - tipped

joints:
  # left front leg
  - lf_shoulder_joint
  - lf_elbow_joint
  - lf_wrist_joint
  # right front leg
  - rf_shoulder_joint
  - rf_elbow_joint
  - rf_wrist_joint
  # left middle leg
  - lm_shoulder_joint
  - lm_elbow_joint
  - lm_wrist_joint
  # right middle leg
  - rm_shoulder_joint
  - rm_elbow_joint
  - rm_wrist_joint
  # left back leg
  - lb_shoulder_joint
  - lb_elbow_joint
  - lb_wrist_joint
  # right back leg
  - rb_shoulder_joint
  - rb_elbow_joint
  - rb_wrist_joint

# static poses
# feet are on the ground evenly and body is standing at mid height
neutral:
  [
    0.0,
    0.1,
    -2.0,
    0.0,
    0.1,
    -2.0,
    0.0,
    0.1,
    -2.0,
    0.0,
    0.1,
    -2.0,
    0.0,
    0.1,
    -2.0,
    0.0,
    0.1,
    -2.0,
  ]

# feet are on the ground and pelvis is nearly touching the ground
sit:
  [
    0.0,
    0.85,
    -2.0,
    0.0,
    0.85,
    -2.0,
    0.0,
    0.85,
    -2.0,
    0.0,
    0.85,
    -2.0,
    0.0,
    0.85,
    -2.0,
    0.0,
    0.85,
    -2.0,
  ]

# limbs off the ground and folded
retracted:
  [
    0.0,
    1.05,
    -2.6,
    0.0,
    1.05,
    -2.6,
    0.0,
    1.05,
    -2.6,
    0.0,
    1.05,
    -2.6,
    0.0,
    1.05,
    -2.6,
    0.0,
    1.05,
    -2.6,
  ]
